api/access/server.go

```diff
@@ -170,12 +173,66 @@ func (s *Server) ExecuteScriptAtBlockHeight(ctx context.Context, in *access.Exec
 	return nil, errors.New("not implemented")
 }
 
-func (s *Server) GetEventsForHeightRange(ctx context.Context, in *access.GetEventsForHeightRangeRequest) (*access.EventsResponse, error) {
-	return nil, errors.New("not implemented")
+func (s *Server) GetEventsForHeightRange(_ context.Context, in *access.GetEventsForHeightRangeRequest) (*access.EventsResponse, error) {
+	var events []*access.EventsResponse_Result
+	for height := in.StartHeight; height <= in.EndHeight; height++ {
+		ee, err := s.index.Events(height)
+		if err != nil {
+			return nil, fmt.Errorf("could not get events at height %d: %w", height, err)
+		}
+
+		header, err := s.index.Header(height)
+		if err != nil {
+			return nil, fmt.Errorf("could not get header at height %d: %w", height, err)
+		}
+
+		result, err := eventsFromBlock(header, ee, header.ID(), height)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse timestamp for block at height %d: %w", height, err)
+		}
+
+		events = append(events, result)
+	}
+
+	resp := access.EventsResponse{
+		Results: events,
+	}
+
+	return &resp, nil
 }
 
-func (s *Server) GetEventsForBlockIDs(ctx context.Context, in *access.GetEventsForBlockIDsRequest) (*access.EventsResponse, error) {
-	return nil, errors.New("not implemented")
+func (s *Server) GetEventsForBlockIDs(_ context.Context, in *access.GetEventsForBlockIDsRequest) (*access.EventsResponse, error) {
+	var events []*access.EventsResponse_Result
+	for _, id := range in.BlockIds {
+		blockID := flow.HashToID(id)
+		height, err := s.index.HeightForBlock(blockID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get height of block with ID %x: %w", id, err)
+		}
+
+		ee, err := s.index.Events(height)
+		if err != nil {
+			return nil, fmt.Errorf("could not get events at height %d: %w", height, err)
+		}
+
+		header, err := s.index.Header(height)
+		if err != nil {
+			return nil, fmt.Errorf("could not get header at height %d: %w", height, err)
+		}
+
+		result, err := eventsFromBlock(header, ee, blockID, height)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse timestamp for block at height %d: %w", height, err)
```

Comment:
This is confusing. It should say could not get events from block or something. Same higher up.